Feature/P-001

Mateusz113 opened the pull request.

Mateusz113 committed:
Create Dockerfile

Mateusz113 committed:
Update pom.xml

Mateusz113 committed:
Create VisitClient.java

Added client for getting visits

Mateusz113 committed:
Create FeignConfig.java

Added config for feign

Mateusz113 committed:
Create MedicalClinicProxyConfig.java

Added clock appwide config

Mateusz113 committed:
Create VisitController.java

Mateusz113 committed:
Update MedicalClinicProxyApplication.java

Added feign annotation

Mateusz113 committed:
Added model classes for visit requests

Mateusz113 committed:
Create VisitService.java

Mateusz113 committed:
Update application.properties

Mateusz113 committed:
Update VisitClient.java

Updated the name
Updated to match the medical clinic visits controller

Mateusz113 committed:
Added filter classes

Added for easy of conversion and communication with actual data owner

Mateusz113 committed:
Create InternalVisitFilterMapper.java

Added mapper for filter class

Mateusz113 committed:
Update application.properties

Added configuration for logging

Mateusz113 committed:
Delete FeignConfig.java

Removed unused config

Mateusz113 committed:
Update VisitController.java

Updated controller to match the medical clinic

Mateusz113 committed:
Update VisitService.java

Simplified methods
Updated to match the client and controller
Used the filter classes

Cymek12 commented on src/main/java/com/mateusz113/medical_clinic_proxy/client/VisitClient.java:
```diff
@@ -0,0 +1,24 @@
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestParam;
+
+@FeignClient(value = "visits-client", url = "medical-clinic:8080/visits")
```
url powinien byc parametryzowany

konradgab commented on pom.xml:
```diff
@@ -46,13 +48,43 @@
+        <dependency>
+            <groupId>org.springframework.cloud</groupId>
+            <artifactId>spring-cloud-starter-openfeign</artifactId>
+            <version>${spring.openfeign.version}</version>
```
wydaje mi sie ze nie jest tutaj potrzebne explicit wskazywanie wersji

konradgab commented on src/main/java/com/mateusz113/medical_clinic_proxy/client/VisitClient.java:
```diff
@@ -0,0 +1,24 @@
+import org.springframework.web.bind.annotation.RequestParam;
+
+@FeignClient(value = "visits-client", url = "medical-clinic:8080/visits")
+public interface VisitClient {
```
MedicalClinicClient

konradgab commented on src/main/resources/application.properties:
```diff
@@ -1,2 +1,12 @@
+logging.level.com.mateusz113.medical_clinic_proxy.client.VisitClient=debug
+# H2 DATABASE CONFIG
+spring.datasource.url=jdbc:h2:mem:public
+spring.datasource.driver-class-name=org.h2.Driver
```
a po co panu konsola h2?

konradgab commented on src/main/resources/application.properties:
```diff
@@ -1,2 +1,12 @@
+logging.level.feign=debug
+logging.level.com.mateusz113.medical_clinic_proxy.client.VisitClient=debug
+# H2 DATABASE CONFIG
+spring.datasource.url=jdbc:h2:mem:public
```
a po co tutaj podlaczenie do bazy danych

konradgab commented on pom.xml:
```diff
@@ -46,13 +48,43 @@
             <version>${mapstruct.version}</version>
         </dependency>
+        <dependency>
+            <groupId>com.h2database</groupId>
```
a po co tutaj w ogole baza danych?

konradgab commented on pom.xml:
```diff
@@ -46,13 +48,43 @@
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
```
a po co tutaj w ogole baza danych?

Mateusz113 committed:
Update pom.xml

Removed unnecessary database dependencies
Added dependency containing common database classes

Mateusz113 committed:
Updated medical clinic client

Updated name of the client
Added parametrized url

Mateusz113 committed:
Update VisitService.java

Replaced client name

Mateusz113 committed:
Update application.properties

Removed database config
Added parametrized url property

Mateusz113 committed:
Update pom.xml

Added oas dependency

Mateusz113 committed:
Update VisitController.java

Added documentation

Mateusz113 committed:
Update application.properties

Added default spingdoc mediaType

Mateusz113 committed:
Update MedicalClinicClient.java

Updated client name and config
Updated endpoint path or registering patients

Mateusz113 committed:
Create CustomErrorDecoder.java

Implemented custom error decoder class

Mateusz113 committed:
Create MedicalClinicClientConfig.java

Implemented config for medical clinic client

Mateusz113 committed:
Update VisitController.java

Added logs for requests

Mateusz113 committed:
Update application.properties

Updated medical clinic client config

Mateusz113 committed:
Update pom.xml

Added resilience circuitbreaker dependency

Mateusz113 committed:
Update MedicalClinicClient.java

Updated medical clinic client config

Mateusz113 committed:
Create MedicalClinicClientFallback.java

Added fallbacks

Mateusz113 committed:
Exceptions and handler setup

Mateusz113 committed:
Update PageableContentDto.java

Added builder

Mateusz113 committed:
Update application.properties

Added circuitbreaker property

Mateusz113 committed:
Update pom.xml

Added WireMock dependency

Mateusz113 committed:
Create MedicalClinicClientTest.java

Added tests for medical clinic client

Mateusz113 committed:
Create application.properties

Added test properties

Mateusz113 committed:
Update VisitController.java

Removed redundant annotation

Mateusz113 committed:
Create VisitTestUtil.java

Added util class

Mateusz113 committed:
Create MedicalClinicProxyIntegrationTest.java

Added integration tests

Mateusz113 committed:
Update MedicalClinicClientTest.java

Removed redundant test
Removed methods now in util class
Added spaces
Removed clock

Mateusz113 committed:
Update VisitTestUtil.java

Separated clock making and getting default time

Mateusz113 committed:
Create VisitServiceTest.java

Mateusz113 committed:
Create VisitControllerTest.java

Mateusz113 committed:
Update MedicalClinicClientTest.java

Updated assertions to use jupiter
fixed patient registering tests bugs

Mateusz113 committed:
Update MedicalClinicClientConfig.java

Overrided default retryer values

Mateusz113 committed:
Update VisitService.java

Added data validation

Mateusz113 committed:
Updated documentation

Added negative responses

Mateusz113 committed:
Added exceptions

Mateusz113 committed:
Update InternalVisitFilter.java

Added builder
